go/fixchain/fix_error.go

```diff
@@ -14,18 +17,20 @@ const (
 	ParseFailure
 	CannotFetchURL
 	FixFailed
+	PostFailed
 	LogPostFailed
 	VerifyFailed
 )
 
 // FixError is the struct with which errors in the fixing process are reported
 type FixError struct {
-	Type  errorType
+	Type  errorType           // The type of error
 	Cert  *x509.Certificate   // The supplied leaf certificate
 	Chain []*x509.Certificate // The supplied chain
 	URL   string              // URL, if a URL is involved
 	Bad   []byte              // The offending bytes, if applicable
```

Comment:
What I mean is that "some day, someone could put something that's not a cert", *that* isn't used, and likely you aren't going to need it (YAGNI). So having it be a narrower interface (a cert) is a good idea, especially as it's usually easier to widen an interface than to retroactively try to narrow it back down!